cli/bblfsh-sdk-tools/cmd/fixtures.go

```diff
@@ -0,0 +1,181 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"gopkg.in/bblfsh/sdk.v1/sdk/driver/integration"
+	"gopkg.in/bblfsh/sdk.v1/manifest"
+	"gopkg.in/bblfsh/sdk.v1/protocol"
+
+	"google.golang.org/grpc"
+)
+
+const FixturesCommandDescription = "" +
+	"Generate integration tests' '.native' and '.uast' fixtures from source files"
+
+type FixturesCommand struct {
+	Args struct {
+		SourceFiles []string `positional-arg-name:"sourcefile(s)" required:"true" description:"File(s) with the source code"`
+	} `positional-args:"yes"`
+	Language      string `long:"language" short:"l" default:"" description:"Language to parse"`
+	Endpoint      string `long:"endpoint" short:"e" default:"localhost:9432" description:"Endpoint of the gRPC server to use"`
+	ExtNative     string `long:"extnative" short:"n" default:"native" description:"File extension for native files"`
+	ExtUast       string `long:"extuast" short:"u" default:"uast" description:"File extension for uast files"`
+	Quiet         bool   `long:"quiet" short:"q" description:"Don't print any output"`
+
+	manifestCommand
+	cli protocol.ProtocolServiceClient
+}
+
+func (c *FixturesCommand) Execute(args []string) error {
+	if err := c.manifestCommand.Execute(args); err != nil {
+		return err
+	}
+
+	c.processManifest(c.Manifest)
+	return nil
+}
+
+func (c *FixturesCommand) processManifest(m *manifest.Manifest) {
+	conn, err := grpc.Dial(c.Endpoint, grpc.WithTimeout(time.Second*2), grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		fmt.Println("Endpoint connection error, is a bblfshd server running?")
+		panic(err)
+	}
+
+	c.cli = protocol.NewProtocolServiceClient(conn)
+
+	for _, f := range c.Args.SourceFiles {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			fmt.Println("Error: File", f, "doesn't exists")
+			os.Exit(1)
+		}
+
+		err := c.generateFixtures(f)
+		if err != nil {
+			fmt.Println("While generating fixtures for ", f)
+			panic(err)
+		}
+	}
+}
+
+func (c *FixturesCommand) generateFixtures(filename string) error {
+	if !c.Quiet {
+		fmt.Println("Processing", filename, "...")
+	}
+
+	source, err := getSourceFile(filename)
+	if err != nil {
+		return err
+	}
+
+	native, err := c.getNative(source, filename)
+	if err != nil {
+		return err
+	}
+
+	err = c.writeResult(filename, native, c.ExtNative)
+	if err != nil {
+		return err
+	}
+
+	uast, err := c.getUast(source, filename)
+	if err != nil {
+		return err
+	}
+
+	err = c.writeResult(filename, uast, c.ExtUast)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *FixturesCommand) getNative(source string, filename string) (string, error) {
+	req := &protocol.NativeParseRequest{
+		Language: c.Language,
+		Content:  source,
+		Filename: removeSourceSuffix(filename),
```

Comment:
This will not work for `java-driver` fixtures. If we're going to use this convention, it would be nice to a) document it somewhere and b) change existing drivers' fixtures to follow this convention.

Reply:
For the Java driver you would need to add the `--language` parameter since they doesn't have any extension that enry can use. I'm documenting the new way of doing integration tests and will add this.